src/CommonLib/LDAPUtils.cs

```diff
@@ -66,6 +65,7 @@ public class LDAPUtils : ILDAPUtils
         private readonly ConcurrentDictionary<string, string> _netbiosCache = new();
         private readonly PortScanner _portScanner;
         private LDAPConfig _ldapConfig = new();
+        private readonly SemaphoreSlim _semaphoreSlim = new(1, 1);
```

Comment:
You have to explicitly specify MaxCount, otherwise there is none defined: https://learn.microsoft.com/en-us/dotnet/api/system.threading.semaphoreslim.-ctor?view=net-8.0#system-threading-semaphoreslim-ctor(system-int32)